### PR TITLE
Add pipeline documentation and examples

### PR DESCRIPTION
**Pull request by jmassardo**
This PR will contain examples and walkthroughs for setting up Chef specific pipelines in several major CI/CD tools.

### CODE REVIEW

**Comment by danielcbright** [comment]
We should definitely work in Policyfiles to these pipeline examples, I'm currently working on one with AWS CodeBuild and will add my example here when I'm done, I think I can use your example @jmassardo from Azure DevOps and add the Policyfile stuff to that as well...

**Comment by danielcbright** [comment]
I've been thinking about this one @jmassardo - with so many pipeline tools out there, maybe we should have the link from the main readme take you to another page that has pipeline tools sorted (Jenkins, Azure DevOps, AWS SSM, etc...) and have the specific examples as links under each of those, that way we can easily keep adding to them.

I'm working on the following btw:

Azure DevOps - Policyfile Pipeline
Azure DevOps - Encrypted Databag Pipeline

Jenkins - Policyfile Pipeline
Jenkins - Encrypted Databag Pipeline

**Comment by jmassardo** [comment]
@danielcbright that's the plan. Each CI/CD tool will have a folder with instructions for each object type along with example code.

**Comment by jmassardo** [comment]
@danielcbright @justinredd Can y'all review the `pipelines/README.md` and the work in the Jenkins folder? If those bits look good, I'll replicate them for ADO and GL.

**Comment by danielcbright** [comment]
> @danielcbright @justinredd Can y'all review the `pipelines/README.md` and the work in the Jenkins folder? If those bits look good, I'll replicate them for ADO and GL.

@jmassardo this looks like a really good first stab, I think it's ready to be pulled in, as long as us as a team agree to use it and make it better over time. Thanks for the time and effort on this, I'm ready to merge when @justinredd gives his 👍

**Comment by jmassardo** [comment]
@danielcbright @justinredd @kevinreedy 

I think this draft is complete. Can y'all test/review and provide feedback? I'd like to get this merged before the end of the week.

File: examples/pipelines/AzureDevOps/cookbook_berks.azure-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  - deployment: Berksfile
+    pool:
+      vmImage: 'ubuntu-latest'
+    environment: 'dev'
+    strategy:
+      runOnce:
+        deploy:
+          steps:
+          - task: vsts-chef-task-install-chefdk@1
+            displayName: 'Install ChefDK'
```
Comment (tas50) [inline examples/pipelines/AzureDevOps/cookbook_berks.azure-pipelines.yml]: Any reason not to use workstation here? DK is going away and is missing a lot of new functionality.

Comment (jmassardo) [reply examples/pipelines/AzureDevOps/cookbook_berks.azure-pipelines.yml]: There isn't a workstation component in the current version of the ADO extension. The Partner team is working on an updated version that includes workstation and addresses some other issues (like full policyfile support).

**Pull request merged**